sdv/data_processing/numerical_formatter.py

```diff
@@ -48,15 +48,25 @@ def __init__(self, learn_rounding_scheme=False, enforce_min_max_values=False,
 
     @staticmethod
     def _learn_rounding_digits(data):
-        # check if data has any decimals
+        """Check if data has any decimals."""
         data = np.array(data)
         roundable_data = data[~(np.isinf(data) | pd.isna(data))]
-        if ((roundable_data % 1) != 0).any():
-            if (roundable_data == roundable_data.round(MAX_DECIMALS)).all():
-                for decimal in range(MAX_DECIMALS + 1):
-                    if (roundable_data == roundable_data.round(decimal)).all():
-                        return decimal
 
+        # Doesn't contain numbers
+        if len(roundable_data) == 0:
+            return None
+
+        # Doesn't contain decimal digits
+        if ((roundable_data % 1) == 0).all():
+            return 0
+
+        # Try to round to fewer digits
+        if (roundable_data == roundable_data.round(MAX_DECIMALS)).all():
+            for decimal in range(MAX_DECIMALS + 1):
+                if (roundable_data == roundable_data.round(decimal)).all():
+                    return decimal
+
+        # Can't round, not equal after MAX_DECIMALS digits of precision
```

Comment:
Can we raise a warning here that informs the user that the rounding could not be detected? Something like ```No rounding scheme detected for column 'interest_rate'. Synthetic data will not be rounded.```

Reply:
Good catch, I completely forgot about the warning message... thanks for reminding me!